app/services/modalService.js

```diff
@@ -26,6 +26,30 @@ angular.module('portainer.services')
     });
   };
 
+  service.prompt = function(options){
+    var box = bootbox.prompt({
+      title: options.title,
+      inputType: options.inputType,
+      inputOptions: options.inputOptions,
+      buttons: {
```

Comment:
A generator function should be used to avoid duplicated code for the generation of the `buttons` object.